mycroft/filesystem/__init__.py

```diff
@@ -31,7 +33,13 @@ def __init__(self, path):
     def __init_path(path):
         if not isinstance(path, str) or len(path) == 0:
             raise ValueError("path must be initialized as a non empty string")
-        path = join(expanduser('~'), '.mycroft', path)
+
+        old_path = join(expanduser('~'), '.mycroft', path)
+        path = join(xdg.BaseDirectory.save_config_path('mycroft'), path)
+
+        # Migrate from the old location if it still exists
+        if isdir(old_path):
+            shutil.move(old_path, path)
```

Comment:
Is there any danger in overwriting data if both paths exist?

Same with the other use of `shutil.move()`

Reply:
it is but I think it's the preferable case...if this move failed someway leaving the two files the move should be performed again since we can't know the state of the target file.

Reply:
should we add a comment regarding when this code will be removed?  is there a release... say 22.02... where we no longer support the old path?

Reply:
Added comment about this

Reply:
There actually was an issue with the move if the target existed as @krisgesling suggested. This has now been handled